lib/url.js

```diff
@@ -498,7 +498,7 @@ Url.prototype.resolveObject = function(relative) {
     if (!relative.host &&
         !/^file:?$/.test(relative.protocol) &&
         !hostlessProtocol[relative.protocol]) {
-      var relPath = (relative.pathname || '').split('/');
+      const relPath = (relative.pathname || '').split('/');
```

Comment:
`relPath` has the potential to be modified.

This should either revert to `var` or be changed to `let`.